In [22]:
import pandas as pd
import numpy as np


# Section 1: Exploratory Data Analysis

### All questions in this section pertain to two of the datasets listed above in Exam Materials 

### (person_level_data.csv and household_level_data.csv), with the data_dictionary.txt file providing the metadata needed to understand the contents and formats of each .csv. 


### Please note that the data is still in its raw, original form and will require some cleaning and/or standardization.


Please write your answers in this document, and include your code as part1_code.txt.
Questions


Please calculate the following summary statistics using the data sets listed above:


1.	Calculate the average weekly hours worked among men and women.
2.	List the three households (i.e. household_ids) with the highest per person incomes. 
3.	List the three individuals (i.e. person_ids) with the highest personal income, and the dollar amounts of their personal incomes.
4.	Calculate the 80th percentile of age among people in each of the five marital categories.


In [23]:
person = pd.read_csv('person_level_data.csv')
household = pd.read_csv('household_level_data.csv')

person.head()

person_id  age citizen_status nativity_status  weekly_hours_worked  \
0         10   62  citizen_birth     native_born                   55   
1         11   65  citizen_birth     native_born                   40   
2         15   70  citizen_birth     native_born                    0   
3         16   70  citizen_birth     native_born                   20   
4         43   75  citizen_birth     native_born                    0   

  marital_status school_status     sex     when_last_worked worked_last_week  \
0        married   not_student  FEMALE         in_last_year           worked   
1        married   not_student       M         in_last_year           worked   
2        married   not_student    MALE  over_five_years_ago     did_not_work   
3        married   not_student       F         in_last_year     did_not_work   
4        married   not_student  FEMALE  over_five_years_ago     not_reported   

   ... race_native_american  race_asian  race_black  race_native_hawaiian  \
0  ...                    0           0           0                     0   
1  ...                    0           0           0                     0   
2  ...                    0           0           0                     0   
3  ...                    0           0           0                     0   
4  ...                    0           0           0                     0   

   race_pacific_islander  race_white  race_other  uninsured  household_id  \
0                      0           1           0          0             5   
1                      0           1           0          0             5   
2                      0           1           0          0             9   
3                      0           1           0          0             9   
4                      0           1           0          0            22   

   hh_income_pct  
0       0.389078  
1       0.610922  
2       0.735381  
3       0.264619  
4       0.250230  

[5 rows x 21 columns]

In [28]:
person.sex.value_counts()

F    10296
M     9601
Name: sex, dtype: int64

In [26]:
person.sex = person.sex.replace({'FEMALE': 'F'})
person.sex = person.sex.replace({'female': 'F'})


In [46]:
person.sex = person.sex.replace({'MALE': 'M'})
person.sex = person.sex.replace({'male': 'M'})
person.sex.value_counts()

F    10296
M     9601
Name: sex, dtype: int64

In [45]:
person.head()

person_id  age citizen_status nativity_status  weekly_hours_worked  \
0         10   62  citizen_birth     native_born                   55   
1         11   65  citizen_birth     native_born                   40   
2         15   70  citizen_birth     native_born                    0   
3         16   70  citizen_birth     native_born                   20   
4         43   75  citizen_birth     native_born                    0   

  marital_status school_status sex     when_last_worked worked_last_week  ...  \
0        married   not_student   F         in_last_year           worked  ...   
1        married   not_student   M         in_last_year           worked  ...   
2        married   not_student   M  over_five_years_ago     did_not_work  ...   
3        married   not_student   F         in_last_year     did_not_work  ...   
4        married   not_student   F  over_five_years_ago     not_reported  ...   

  race_native_american  race_asian  race_black  race_native_hawaiian  \
0                    0           0           0                     0   
1                    0           0           0                     0   
2                    0           0           0                     0   
3                    0           0           0                     0   
4                    0           0           0                     0   

   race_pacific_islander  race_white  race_other  uninsured  household_id  \
0                      0           1           0          0             5   
1                      0           1           0          0             5   
2                      0           1           0          0             9   
3                      0           1           0          0             9   
4                      0           1           0          0            22   

   hh_income_pct  
0       0.389078  
1       0.610922  
2       0.735381  
3       0.264619  
4       0.250230  

[5 rows x 21 columns]

In [56]:
f_df = person.loc[person.sex == 'F']

f_df.weekly_hours_worked.mean()

m_df = person.loc[person.sex == 'M']

m_df.weekly_hours_worked.mean()

28.66993021560254

## Calculate the average weekly hours worked among men and women.

### On average men worked 28.67 hours

### On average women worked 21.73 hours

## List the three households (i.e. household_ids) with the highest per person incomes.

In [58]:
household.head()

household_id      variable    value
0             5     hh_income  90334.0
1             5       hh_size      2.0
2             5  num_children      0.0
3             9     hh_income  66216.0
4             9       hh_size      2.0

In [60]:
household.variable.value_counts()

num_children    10986
hh_income       10986
hh_size         10986
Name: variable, dtype: int64

In [98]:
income_df = household.loc[household.variable == 'hh_income']

income_df.sort_values(by='value', ascending = False)


household_id   variable      value
14892         24934  hh_income  1115407.0
5955          10110  hh_income  1109371.0
23934         40137  hh_income  1040447.0
17223         28981  hh_income  1010268.0
6669          11208  hh_income  1001694.0
...             ...        ...        ...
27429         46013  hh_income        1.0
3090           5092  hh_income        1.0
29220         49062  hh_income        1.0
20586         34684  hh_income        1.0
4017           6779  hh_income        1.0

[10986 rows x 3 columns]

In [99]:
size_df = household.loc[household.variable == 'hh_size']
size_df.sort_values(by='value', ascending = False)

household_id variable  value
5962          10118  hh_size   12.0
5884          10014  hh_size   12.0
18175         30623  hh_size   11.0
24013         40285  hh_size   11.0
18919         31927  hh_size   10.0
...             ...      ...    ...
9532          15824  hh_size    1.0
22804         38320  hh_size    1.0
22810         38327  hh_size    1.0
22813         38330  hh_size    1.0
32956         55148  hh_size    1.0

[10986 rows x 3 columns]

In [ ]:
joined = income_df.join

###  Top three household income household_id 24934, 10110, 40137

In [91]:
household.head()

household_id      variable    value
0             5     hh_income  90334.0
1             5       hh_size      2.0
2             5  num_children      0.0
3             9     hh_income  66216.0
4             9       hh_size      2.0

In [96]:
household['per_pers

KeyError: 'hh_size'